pypeman/channels.py

```diff
@@ -498,19 +502,21 @@ class SubChannel(BaseChannel):
     """ Subchannel used for forking channel processing. """
 
     def callback(self, fut):
+        self.parent.sub_chan_tasks.remove(fut)
         try:
             result = fut.result()
             logger.debug("Subchannel %s end process message %s", repr(self), result)
         except Dropped:
             self.logger.info("Subchannel %s. Msg was dropped", repr(self))
         except Exception:
             self.logger.exception("Error while processing msg in subchannel %s", self)
+            raise
```

Comment:
Is there any possibility. that the rise here will cause some potentially additional code to be not executed.


Example.

Let's assume callbacks are caled in following way.

```
for callback in [callback1, vallback2, callback3]:
    callback()
```

With the now introduced `raise` callback2 and callback3 would not be called if callback1 had an an exceptio

Reply:
As I teld you, `asyncio.gather()` don't block other tasks if the first have an exception
+ The subchannel handle function and his callback are launched in the "finally" part of the parent chan handler() so when himself have finished to process the message
I add another subchannel in test in case of